File: archive/j/java/BubbleSort.java
import java.util.Scanner;

class BubbleSort {

    public static void main(String[] args) {
        int c, d, swap;
        String inputString = "";
        Scanner in = new Scanner(System.in);
        System.out.println("Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"");
        inputString = in.nextLine();
        inputString = inputString.trim();
        String[] unsortedString = new String[inputString.length()];

        if (inputString.matches("[\\d,]+")) {

            unsortedString = inputString.split(",");
            System.out.println(unsortedString[0] + "," + unsortedString[2]);

            int size = unsortedString.length;
            int[] arr = new int[size];
            for (int i = 0; i < size; i++) {
                arr[i] = Integer.parseInt(unsortedString[i]);
                System.out.println(arr[i]);
            }
            for (c = 0; c < (arr.length - 1); c++) {
                for (d = 0; d < arr.length - c - 1; d++) {
                    if (arr[d] > arr[d + 1]) {
                        swap = arr[d];
                        arr[d] = arr[d + 1];
                        arr[d + 1] = swap;
                    }
                }
            }

            for (c = 0; c < arr.length; c++) {
                System.out.print(arr[c]);
                if (c == arr.length - 1) {
                } else {
                    System.out.print(",");
                }
            }
        } else {

            System.out.println("Usage: please provide a list of at least two integers to sort in the format \"1, 2, 3, 4, 5\"");

        }
    }
}

File: archive/j/javascript/bubble-sort.js
/**
 * Bubble sort algorithm
 *
 * @param {Integer array} arr
 */

function bubblesort(arr) {
    let swap;

    // repeat while array is unsorted
    do {
        swap = false;

        // iterate array
        for (let index = 0; index < arr.length - 1; index++) {
            // if current item is out of order
            if (arr[index] > arr[index + 1]) {
                // hotswap the current element with the next element
                [arr[index], arr[index + 1]] = [arr[index + 1], arr[index]];

                // flag that list was unordered in this pass
                swap = true;
            }
        }

    } while (swap);

    return arr;
}

/**
 * Executable function
 * 
 * @param {Command line input} input
 */

function main(input) {
    // usage text
    const usage = `Usage: please provide a list of at least two integers to sort in the format “1, 2, 3, 4, 5”`;

    /**
     * If the string matches the format `"[number], [number], (... [number])"`
     * we have a valid input.
     */
    const inputValidation = /^(\d+,\s*){2,}$/m;

    if (inputValidation.test(input)) {
        // valid input
        let arr;

        /**
         * Format string to be bubblesorted.
         *  - strip all whitespace
         *  - split into array at ',' character
         *  - convert all elements to integers
         *  - filter out NaN elements (uncaught text characters or empty elements)
         */
        arr = input.replace(/\s/g, '');
        arr = arr.split(',');
        arr = arr.map(function (n) {
            return parseInt(n, 10);
        });
        arr = arr.filter(n => n);

        // apply bubblesort and output result
        console.log(bubblesort(arr));
    }
    else {
        // invalid input
        console.log(usage);
    }
}

// run the executable function
const input = process.argv[2];
main(input);
